modules/assim.sequential/R/sda.enkf.R

```diff
@@ -274,454 +284,516 @@ sda.enkf <- function(settings, obs.mean, obs.cov, IC = NULL, Q = NULL){
+        mu.a <- colMeans(dat[, iX])
+        Pa   <- cov(dat[, iX])
+        Pa[is.na(Pa)] <- 0
+        
+        CI.X1[, t] <- quantile(dat[, iX[1]], c(0.025, 0.5, 0.975))
+        CI.X2[, t] <- quantile(dat[, iX[2]], c(0.025, 0.5, 0.975))
+        
+        mq <- dat[, iq]  # Omega, Precision
+        q.bar <- matrix(apply(mq, 2, mean), length(mu.f), length(mu.f))  # Mean Omega, Precision
+        
+        col <- matrix(1:length(mu.f) ^ 2, length(mu.f), length(mu.f))
+        WV  <- matrix(0, length(mu.f), length(mu.f))
+        for (i in seq_along(mu.f)) {
+          for (j in seq_along(mu.f)) {
+            WV[i, j] <- wish.df(q.bar, X = mq, i = i, j = j, col = col[i, j])
+          }
+        }
+        
+        n <- mean(WV)
+        if (n < length(mu.f)) {
+          n <- length(mu.f)
+        }
+        V <- solve(q.bar) * n
+        
+        aqq[t + 1, , ]   <- V
+        bqq[t + 1]       <- n
+        enkf.params[[t]] <- list(mu.f = mu.f, Pf = Pf, mu.a = mu.a, 
+                                 Pa = Pa, q.bar = q.bar, n = n)
       }
       
-      n = mean(WV)
-      if(n < length(mu.f)) n = length(mu.f)
-      V = solve(q.bar)*n
-      
-      aqq[t+1,,] = V
-      bqq[t+1] = n
-      enkf.params[[t]] <- list(mu.f = mu.f, Pf = Pf, mu.a = mu.a, Pa = Pa, q.bar=q.bar, n=n)
-    }
-    
     } else {
       ###-------------------------------------------------------------------###
       ### No Observations -- Starts Here                                    ###
       ###-------------------------------------------------------------------### 
       
       ### no process variance -- forecast is the same as the analysis ###
-      if(processvar==FALSE){
-        mu.a = mu.f
-        Pa = Pf + Q
-      ### yes process variance -- no data 
-      }else{
-        mu.a = mu.f
-        Pa = Pf + solve(q.bar)
+      if (!processvar) {
+        mu.a <- mu.f
+        Pa   <- Pf + Q
+        ### yes process variance -- no data
+      } else {
+        mu.a <- mu.f
+        Pa   <- Pf + solve(q.bar)
       }
       enkf.params[[t]] <- list(mu.f = mu.f, Pf = Pf, mu.a = mu.a, Pa = Pa)
     }
     
     ## update state matrix
-    analysis <- as.data.frame(rmvnorm(as.numeric(nens),mu.a,Pa,method="svd"))
+    analysis <- as.data.frame(rmvnorm(as.numeric(nens), mu.a, Pa, method = "svd"))
     colnames(analysis) <- colnames(X)
     
     ## in the future will have to be separated from analysis
-    new.state <- analysis
+    new.state  <- analysis
     new.params <- params
     
-    ANALYSIS[[t]] = analysis
+    ANALYSIS[[t]] <- analysis
     
-    if(interactive() & t>1){
-      t1=1
-      names.y <- unique(unlist(lapply(obs.mean[t1:t],function(x){return(names(x))})))
-      Ybar = t(sapply(obs.mean[t1:t],function(x){
-        tmp <- rep(NA,length(names.y))
-        names(tmp) <- names.y 
-        mch = match(names(x),names.y)
-        tmp[mch] = x[mch]
-        return(tmp)
+    if (interactive() & t > 1) {
```

Comment:
No, it's not for debugging and yes, it belongs there.  If someone is running sda.enkf from an interactive terminal then the code updates the progress of the enkf time series plot every iteration. This provides a real time update of the code's progress and performance. However, we don't want to generate the same plots if someone submits runs the workflow in batch mode or submits it to a queue.